compiler/plugins/target/CUDA/CUDATarget.cpp

```diff
@@ -55,59 +55,61 @@ namespace mlir::iree_compiler::IREE::HAL {
-    binder.opt<bool>("iree-hal-cuda-dump-ptx", dumpPtx, llvm::cl::cat(category),
+    binder.opt<bool>("iree-cuda-dump-ptx", dumpPtx, llvm::cl::cat(category),
                      llvm::cl::desc("Dump ptx to the debug stream."));
 
-    binder.opt<std::string>("iree-hal-cuda-llvm-target-arch", clTargetChip,
-                            llvm::cl::cat(category),
-                            llvm::cl::desc("LLVM target chip."));
+    binder.opt<std::string>(
+        "iree-cuda-target", clTarget, llvm::cl::cat(category),
+        llvm::cl::desc("CUDA target as expected by LLVM NVPTX backend; e.g., "
+                       "'sm_80'/'sm_90' for targeting Ampere/Hopper GPUs. "
+                       "Additionally this also supports architecture code "
+                       "names like 'turing'/'ampere' or some product names "
+                       "like 'a100'/'rtx3090ti' for a better experience. See "
+                       "https://iree.dev/guides/deployment-configurations/"
+                       "gpu-cuda for more details."));
 
-    binder.opt<std::string>("iree-hal-cuda-llvm-target-feature",
-                            clTargetFeature, llvm::cl::cat(category),
-                            llvm::cl::desc("Use to set PTX version."));
+    binder.opt<std::string>(
+        "iree-cuda-target-features", clTargetFeatures, llvm::cl::cat(category),
+        llvm::cl::desc(
+            "CUDA target features as expecgted by LLVM NVPTX backend; e.g., "
+            "use '+ptxNN' to set PTX version to NN."));
 
     binder.opt<bool>(
-        "iree-hal-cuda-use-ptxas", clUsePtxas, llvm::cl::cat(category),
-        llvm::cl::desc("It uses the ptxas compiler that is on the environment, "
-                       "compiles the "
-                       "generated PTX code with it, puts the cubin binary "
-                       "generated by ptxas "
-                       "into the executable. "
-                       "'--iree-hal-cuda-llvm-target-arch' is used as "
-                       "the target GPU. If passing additional parameters to "
-                       "ptxas is desired, "
-                       "the parameters flag can be used "
-                       "(e.g.'--iree-hal-cuda-use-ptxas-params=-v')."));
+        "iree-cuda-use-ptxas", clUsePtxas, llvm::cl::cat(category),
+        llvm::cl::desc(
+            "It uses the ptxas compiler that is on the environment, compiles "
+            "the generated PTX code with it, puts the cubin binary generated "
+            "by ptxas into the executable. '--iree-cuda-target' is used as the "
+            "target GPU. If passing additional parameters to ptxas is desired, "
+            "the parameters flag can be used "
+            "(e.g.'--iree-cuda-use-ptxas-params=-v')."));
 
     binder.opt<std::string>(
-        "iree-hal-cuda-use-ptxas-from", clUsePtxasFrom, llvm::cl::cat(category),
+        "iree-cuda-use-ptxas-from", clUsePtxasFrom, llvm::cl::cat(category),
```

Comment:
(Preexisting code, so could ignore this)

Is this `iree-cuda-use-ptxas-from` option compatible with `iree-cuda-use-ptxas`? The options could be structured so they are mutually exclusive: one searches the environment and one uses an explicit path.

Reply:
`iree-cuda-use-ptxas-from` and `iree-cuda-use-ptxas-params` are additional options we can pass in if `iree-cuda-use-ptxas` is set.

Reply:
Made it clear by updating the doc.